Reaqtive/Core/Reaqtive.Linq/Reaqtive/Operators/DefaultIfEmpty.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT License.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Reaqtive.Operators
+{
+    internal sealed class DefaultIfEmpty<TSource> : SubscribableBase<TSource>
+    {
+        private readonly ISubscribable<TSource> _source;
+
+        public DefaultIfEmpty(ISubscribable<TSource> source)
```

Comment:
We should also have the variant that takes in a `TSource` with the default value to emit. There are two overloads:

```csharp
IO<T> DefaultIfEmpty<T>(IO<T>)
IO<T> DefaultIfEmpty<T>(IO<T>, T)
```

We can implement the former in terms of the latter by just using `default(T)` as the default value to emit.